engine.go

```diff
@@ -1,16 +1,139 @@
 package fuego
 
-func NewEngine() *Engine {
-	return &Engine{
-		OpenAPI:      NewOpenAPI(),
-		ErrorHandler: ErrorHandler,
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"log/slog"
+	"os"
+	"path/filepath"
+)
+
+func NewEngine(config ...OpenAPIConfig) *Engine {
+	if len(config) > 1 {
+		panic("config should not be more than one")
+	}
+	engine := &Engine{
+		OpenAPI:       NewOpenAPI(),
+		OpenAPIConfig: defaultOpenAPIConfig,
+		ErrorHandler:  ErrorHandler,
+	}
+	if len(config) > 0 {
+		engine.setOpenAPIConfig(config[0])
 	}
+	return engine
 }
 
 // The Engine is the main struct of the framework.
 type Engine struct {
-	OpenAPI      *OpenAPI
-	ErrorHandler func(error) error
+	OpenAPI       *OpenAPI
+	ErrorHandler  func(error) error
+	OpenAPIConfig OpenAPIConfig
 
 	acceptedContentTypes []string
 }
+
+type EngineOpenAPIConfig struct {
+	// If true, the engine will not print messages
+	DisableMessages bool
+	// If true, the engine will not save the OpenAPI JSON spec locally
+	DisableLocalSave bool
+	// Local path to save the OpenAPI JSON spec
+	JsonFilePath string
+	// Pretty prints the OpenAPI spec with proper JSON indentation
+	PrettyFormatJson bool
+}
+
+// OutputOpenAPISpec takes the OpenAPI spec and outputs it to a JSON file
+func (e *Engine) OutputOpenApiSpec() []byte {
+	e.OpenAPI.computeTags()
+
+	// Validate
+	err := e.OpenAPI.Description().Validate(context.Background())
+	if err != nil {
+		slog.Error("Error validating spec", "error", err)
+	}
+
+	// Marshal spec to JSON
+	jsonSpec, err := e.marshalSpec()
+	if err != nil {
+		slog.Error("Error marshaling spec to JSON", "error", err)
+	}
+
+	if !e.OpenAPIConfig.DisableLocalSave {
+		err := e.saveOpenAPIToFile(e.OpenAPIConfig.JsonFilePath, jsonSpec)
+		if err != nil {
+			slog.Error("Error saving spec to local path", "error", err, "path", e.OpenAPIConfig.JsonFilePath)
+		}
+	}
+	return jsonSpec
+}
+
+func (e *Engine) saveOpenAPIToFile(jsonSpecLocalPath string, jsonSpec []byte) error {
+	jsonFolder := filepath.Dir(jsonSpecLocalPath)
+
+	err := os.MkdirAll(jsonFolder, 0o750)
+	if err != nil {
+		return errors.New("error creating docs directory")
+	}
+
+	f, err := os.Create(jsonSpecLocalPath) // #nosec G304 (file path provided by developer, not by user)
+	if err != nil {
+		return errors.New("error creating file")
+	}
+	defer f.Close()
+
+	_, err = f.Write(jsonSpec)
+	if err != nil {
+		return errors.New("error writing file ")
+	}
```

Comment:
Two issues here

- a strange trailing space
- you create an error, whole you have already one: wrap it


```suggestion
	_, err = f.Write(jsonSpec)
	if err != nil {
		return fmt.Errorf("error writing file: %w", err)
	}
```

Reply:
So I'm happy to do this. This PR was more just moving this code to this file and changing the target of method to be the `Engine`. I'd like to handle this in a separate PR or you're welcome too.

Reply:
Resolving to be dealt with in another PR